spark-gremlin/src/test/java/org/apache/tinkerpop/gremlin/hadoop/structure/io/AbstractIoRegistryCheck.java

```diff
@@ -72,7 +72,7 @@ public void checkGryoV1d0IoRegistryCompliance(final HadoopGraph graph, final Cla
         graph.configuration().setProperty(IoRegistry.IO_REGISTRY, ToyIoRegistry.class.getCanonicalName());
         final GryoRecordWriter writer = new GryoRecordWriter(new DataOutputStream(new FileOutputStream(input)), ConfUtil.makeHadoopConfiguration(graph.configuration()));
         validateIoRegistryGraph(graph, graphComputerClass, writer);
-        assertTrue(input.delete());
+        input.deleteOnExit();
```

Comment:
I am wondering if it would be better to check OS and run either `.delete()` if Unix or `.deleteOnExit()` if Windows. It would maintain the assertion check for Unix but also allow Windows build to run as expected. `.deleteOnExit()` returns void.

Reply:
Let me know what you all think

Reply:
I think converting to deleteOnExit() is okay. Let's not put OS specific stuff into the code since that is very against the ideology of Java.

Reply:
Why does it fail on Windows would be the first question.

Reply:
PR comment ^

Reply:
@L0Lmaker Could we call close on the file then delete?

Reply:
`java.io.File` doesn't represent an open file, it represents a path in the filesystem. Therefore there isn't a `close()` method on `File`

Reply:
Hah, what a conundrum. Okay your solution LGTM.